packages/app-page-builder/src/editor/plugins/elements/pagesList/PagesList.tsx

```diff
@@ -64,6 +65,9 @@ const PagesList = props => {
 
     const listPublishedPages = get(data, "pageBuilder.listPublishedPages");
 
+    /**
+     * How to handle these two checks if `meta` field no longer has `previousPage` and `nextPage` fields?
```

Comment:
Good question @dibenso 👍 

In the new schema, we'll use `meta.hasMoreItems` to check whether there are more items
and `meta.cursor` to fetch more items by passing it as `after` variable in the GQL query.

One way to implement pagination is to keep a list of such cursors and then maintain an index to paginate through pages i.e. add the new cursor to the list and increment the index or page for `nextPage` and decrement the index for `prevPage`. For example:

```js
 const [cursors, setCursors] = useState([null]);
 const [page, setPage] = useState(0);

// Pass cursor as after argument
 const variables = {
        sort,
        where: {
            category: vars.category,
            tags: {
                query: vars.tags,
                rule: vars.tagsRule
            }
        },
        limit: parseInt(vars.resultsPerPage),
        after: cursors[page],
        exclude: [pageAtomValue.path]
    };

// And then, paginate using the cursor and pseudo page

let prevPage = null;
    if (page >= 1) {
        prevPage = () => {
            setPage(page => page - 1);
        };
    }

    let nextPage = null;
    if (listPublishedPages.meta.cursor) {
        nextPage = () => {
            setCursors(cursors => [...cursors, listPublishedPages.meta.cursor]);
            setPage(page => page + 1);
        };
    }

```